app/src/main/java/com/my/journeyplanner/MainActivity.kt

```diff
@@ -1,12 +1,68 @@
 package com.my.journeyplanner
 
 import android.os.Bundle
+import android.view.KeyEvent
+import android.view.inputmethod.EditorInfo
+import android.widget.Button
+import android.widget.EditText
 import androidx.appcompat.app.AppCompatActivity
+import com.my.journeyplanner.helpers.JourneyPlannerApiService
+import com.my.journeyplanner.models.JourneyPlannerDisambiguationResult
+import com.my.journeyplanner.presenters.MainPresenter
+import com.my.journeyplanner.views.main.MainContract
+import retrofit2.Call
 
-class MainActivity : AppCompatActivity() {
+val apiService by lazy { JourneyPlannerApiService.createApiService }
```

Comment:
what is the reason why this is lazy. Why did you put this here

Reply:
Whoops!. Shouldn't be lazy as it's needed all the time. Moved back to presenter as advised in  later comment